### PR TITLE
VSTSCLTLoadAgentsTemplate

### PR DESCRIPTION
**Pull request by dpksinghal**
Adding the VM template for VSTS load agent.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dpksinghal__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (dsinghal). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by dpksinghal** [comment]
I am getting the following error in CI failure logs.

Server Error:{
"error": {
"code": "ENOTFOUND",
"errno": "ENOTFOUND",
"syscall": "getaddrinfo",
"hostname": "arm-validator.40.118.211.57.xip.io"
}
}

In old threads/issues, I found that it is some intermittent issue:
[https://github.com/Azure/azure-quickstart-templates/pull/1980]

How shall I restart the CI build?

File: 301-vsts-clt-loadagentrig/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,217 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+  "VSTSAccountName": {
```
Comment (singhkay) [inline 301-vsts-clt-loadagentrig/azuredeploy.json]: @dpksinghal Please use camelCase on parameter names

File: 301-vsts-clt-loadagentrig/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      "type": "string",
+      "metadata": {"description": "Please enter the VSTS account name without .visualstudio.com, e.g., 'xyz' is VSTS account name for 'xyz.visualstudio.com'"}
+    },
+    "PATToken": {
```
Comment (singhkay) [inline 301-vsts-clt-loadagentrig/azuredeploy.json]: This file could use formatting to improve readability

File: 301-vsts-clt-loadagentrig/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    },
+    "vmCount": {
+      "type": "int",
+      "metadata": { "description": "Number of virtual machines" },
+	    "defaultValue": 1
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": { "description": "User name for an account on virtual machine(s)." },
+	  "defaultValue": "cltuser"
```
Comment (singhkay) [inline 301-vsts-clt-loadagentrig/azuredeploy.json]: Please remove the default value as a security best practice

File: 301-vsts-clt-loadagentrig/metadata.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "VSTS-CLT-LoadAgent-VMRig",
+  "description": "This template can be used to provision azure VMs to run load test using VSTS CLT service.",
+  "summary": "Yet to come",
+  "githubUsername": "dpksinghal",
+  "dateUpdated": "<e.g. 2015-12-20>"
```
Comment (singhkay) [inline 301-vsts-clt-loadagentrig/metadata.json]: Can you enter the actual date?

File: 301-vsts-clt-loadagentrig/metadata.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "VSTS-CLT-LoadAgent-VMRig",
```
Comment (singhkay) [inline 301-vsts-clt-loadagentrig/metadata.json]: Can you enter a better name for display on Azure.com

File: 301-vsts-clt-loadagentrig/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+        "publisher": "Microsoft.Compute",
+        "type": "CustomScriptExtension",
+        "forceUpdateTag": "[variables('sequenceVersion')]",
+        "typeHandlerVersion": "1.8",
+        "autoUpgradeMinorVersion": true,
+        "settings": {
+          "fileUris": [
+            "https://elsprodch2su1.blob.core.windows.net/ets-containerfor-loadagentresources/bootstrap/ConfigureUserLoadAgent.ps1"
+          ],
+		  "commandToExecute":"[concat('powershell.exe -ExecutionPolicy Unrestricted -File .\\bootstrap\\ConfigureUserLoadAgent.ps1 -TeamServicesAccountUrl https://', parameters('VSTSAccountName'), '.vsclt.visualstudio.com', ' -PATToken ', parameters('PATToken'))]"
```
Comment (singhkay) [inline 301-vsts-clt-loadagentrig/azuredeploy.json]: PAT Token seems like something secure that shouldn't be logged on the server and returned in the GET VM call. In that case pass commandToExecute as part of protected settings.

**Pull request merged**